### PR TITLE
Enable SLES11 OS Image Build Host

### PR DESCRIPTION
**Pull request by nadvornik**
## What does this PR change?

This pull request enables building of SLES11 images, including i586 architecture.

- enable SLES11 build host
- python 2.6 compatibility fixes
- adjust Kiwi commandline and inspection scripts to work with SLES11 kiwi
- add a workaround for creating bundle tarball with SLES11 kiwi
- add a workaround for repo URLs problem with SLES11 kiwi
- generate SLE11 specific ssl-cert-osimage package

## GUI diff

No difference.

## Documentation

TBD.

- [ ] **DONE**

## Test coverage

TBD.

- [ ] **DONE**

## Links

https://github.com/SUSE/spacewalk/issues/7471

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

File: spacewalk/certs-tools/mgr-package-rpm-certificate-osimage [hunk 1]
```diff
@@ -47,8 +49,12 @@ def processCommandline():
 
 def genCaRpm(options):
     OSIMAGE_RPM_CERTIFICATE_PATH = "/usr/share/susemanager/salt/images"
-    CA_CERT_RPM_NAME_OSIMAGE = CA_CRT_RPM_NAME + "-osimage"
-    OSIMAGE_RPM_REQUIRES = "ca-certificates"
+    if options.target_os == 'SLE11':
+        CA_CERT_RPM_NAME_OSIMAGE = CA_CRT_RPM_NAME + "-osimage-sle11"
+        OSIMAGE_RPM_REQUIRES = ["openssl-certs", "coreutils"]
```
Comment (mbologna) [inline spacewalk/certs-tools/mgr-package-rpm-certificate-osimage]: `openssl-certs` is the equivalent in SLES11 for `ca-certificates`?

File: spacewalk/config/etc/httpd/conf.d/zz-spacewalk-www.conf [hunk 2]
```diff
@@ -52,6 +52,10 @@ RewriteRule ^/WEBRPC /rhn/rpc/api
 RewriteRule ^/rhn/manager/download/dists/(.*)/main/(.*)/(Packages.*)$ /rhn/manager/download/$1/repodata/$3
 RewriteRule ^/rhn/manager/download/dists/(.*)/(InRelease|Release.*)$ /rhn/manager/download/$1/repodata/$2
 
+# Workaround for SLE11 Kiwi that unconditionally appends extra params to https urls
```
Comment (mbologna) [inline spacewalk/config/etc/httpd/conf.d/zz-spacewalk-www.conf]: Why only for SLE11?

File: susemanager-utils/susemanager-sls/salt/images/kiwi-image-build.sls [hunk 3]
```diff
@@ -35,20 +45,24 @@ mgr_buildimage_prepare_activation_key_in_source:
...
     - require:
       - cmd: mgr_buildimage_kiwi_prepare
 
 mgr_buildimage_kiwi_bundle:
   cmd.run:
-    - name: "kiwi --nocolor --yes --bundle-build {{ dest_dir }} --bundle-id {{ bundle_id }} --destdir {{ bundle_dir }} {{ kiwi_params }}"
+{%- if have_bundle_build %}
+    - name: "{{ kiwi }} --nocolor --yes --bundle-build {{ dest_dir }} --bundle-id {{ bundle_id }} --destdir {{ bundle_dir }}"
+{%- else %}
+    - name: "mkdir -p '{{ bundle_dir }}' && cd '{{ dest_dir }}' && tar czf '{{ bundle_dir }}'`basename *.packages .packages`-{{ bundle_id }}.tgz --no-recursion `find . -maxdepth 1 -type f` && cd '{{ bundle_dir }}' && sha256sum *.tgz > `echo *.tgz`.sha256"
```
Comment (mbologna) [inline susemanager-utils/susemanager-sls/salt/images/kiwi-image-build.sls]: Can you please insert a comment on why do we need this bundle?

File: susemanager-utils/susemanager-sls/src/modules/kiwi_info.py [hunk 4]
```diff
@@ -37,6 +37,22 @@ def parse_buildinfo(dest):
                 group[match.group('name')] = match.group('val')
     return ret
 
+def guess_buildinfo(dest):
```
Comment (mbologna) [inline susemanager-utils/susemanager-sls/src/modules/kiwi_info.py]: Is this related to SLE11?

File: susemanager-utils/susemanager-sls/src/modules/kiwi_source.py [hunk 5]
```diff
@@ -100,7 +100,7 @@ def _prepareGit(source, dest, root):
   if rev == '':
     rev = 'master'
 
-  log.debug('GIT URL: {}, Revision: {}, subdir: {}'.format(url, rev, subdir))
+  log.debug('GIT URL: {0}, Revision: {1}, subdir: {2}'.format(url, rev, subdir))
```
Comment (mbologna) [inline susemanager-utils/susemanager-sls/src/modules/kiwi_source.py]: Specifying placeholders index is mandatory for Python 2.6. Is this intended?

**Comment by mbologna** [comment]
I almost understood all the changes you made (thanks for taking care of this) but I think it needs some documentation (I suggest to add some comments in the code).
Please also adjust the pull request template above.

File: susemanager-utils/susemanager-sls/salt/images/kiwi-image-build.sls [hunk 6]
```diff
@@ -13,8 +13,18 @@
 {%- set bundle_dir = root_dir + '/images/' %}
 {%- set bundle_id  = pillar.get('build_id') %}
 {%- set activation_key = pillar.get('activation_key') %}
+{%- set kiwi = 'kiwi' %}
```
Comment (aaannz) [inline susemanager-utils/susemanager-sls/salt/images/kiwi-image-build.sls]: Why do you set kiwi here when later you reassign it anyway?

File: susemanager-utils/susemanager-sls/salt/images/kiwi-image-build.sls (hunk 3 above)
Comment (aaannz) [reply susemanager-utils/susemanager-sls/salt/images/kiwi-image-build.sls]: We need the bundle because image collect is based upon assumption we get only one file from build host which is tgz with both system and boot images.
But this oneliner is quite massive. What do you think about moving the `{%- if have_bundle_build %}` condition around whole `mgr_buildimage_kiwi_bundle` state and prepare different one with multiple `cmd.run` states in it?

File: susemanager-utils/susemanager-sls/salt/services/kiwi-image-server.sls [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Image Server installation state - part of SUSE Manager for Retail
 #
-# Copyright © 2017, 2018 SUSE LLC
+# Copyright (c) 2017, 2018 SUSE LLC
```
Comment (aaannz) [inline susemanager-utils/susemanager-sls/salt/services/kiwi-image-server.sls]: Can you also update a year when you are at it? And also update year in all here modified files?

File: susemanager-utils/susemanager-sls/src/modules/kiwi_info.py [hunk 8]
```diff
@@ -37,6 +37,22 @@ def parse_buildinfo(dest):
...
+    files = __salt__['file.readdir'](dest)
+
+    pattern_basename = re.compile(r"^(?P<basename>.*)\.packages$")
+    pattern_pxe = re.compile(r"^initrd-netboot")
+    for f in files:
+        match = pattern_basename.match(f)
+        if match:
+            ret['main']['image.basename'] = match.group('basename')
+
+        match = pattern_basename.match(f)
```
Comment (aaannz) [inline susemanager-utils/susemanager-sls/src/modules/kiwi_info.py]: This should be `match = pattern_pxe.match(f)` right?

File: susemanager-utils/susemanager-sls/src/modules/kiwi_source.py (hunk 5 above)
Comment (aaannz) [reply susemanager-utils/susemanager-sls/src/modules/kiwi_source.py]: I guess this is because of SLE11.

File: spacewalk/certs-tools/mgr-package-rpm-certificate-osimage (hunk 1 above)
Comment (nadvornik) [reply spacewalk/certs-tools/mgr-package-rpm-certificate-osimage]: The post-install script on SLES11 calls binaries from openssl-certs and coreutils to deploy the certificate:
https://github.com/uyuni-project/uyuni/blob/master/spacewalk/certs-tools/update-ca-cert-trust.sh

File: spacewalk/config/etc/httpd/conf.d/zz-spacewalk-www.conf (hunk 2 above)
Comment (nadvornik) [reply spacewalk/config/etc/httpd/conf.d/zz-spacewalk-www.conf]: This is some feature/bug in SLES11 kiwi, probably related to old NCC. It is safer to fix it here than touching the kiwi package.
SLES12 Kiwi does not need it.

File: susemanager-utils/susemanager-sls/src/modules/kiwi_info.py (hunk 4 above)
Comment (nadvornik) [reply susemanager-utils/susemanager-sls/src/modules/kiwi_info.py]: Yes, SLES11 Kiwi does not create the buildinfo file, so the information has to be extracted this way.

File: susemanager-utils/susemanager-sls/src/modules/kiwi_source.py (hunk 5 above)
Comment (nadvornik) [reply susemanager-utils/susemanager-sls/src/modules/kiwi_source.py]: Yes, this is needed because SLES11 has python 2.6

**Comment by juliogonzalez** [comment]
spacewalk-config did not have a changelog entry. Adding it now.
